brave/brave-ios
# Fix #5787: Send to ENS wallet address

## Summary of Changes
- Support ENS wallet address resolution in Send token
- Includes ENS offchain wallet address resolution, which defaults to asking the user before submitting request offchain.
- Error copy was updated for SNS and ENS to be dynamic: https://github.com/brave/brave-browser/issues/28041
    - ex. `Domain doesn't have a linked Ethereum address`
- Security / privacy review: https://github.com/brave/security/issues/1206 

This pull request fixes brave/brave-ios#5787

## Submitter Checklist:


## Test Plan:
1. Open wallet/web3 settings and verify default setting of `Allow ENS Offchain Lookup` is set to `Ask`
2. Navigate to send token view
3. Select an Ethereum network
4. Enter `ethereum.eth` as a send to address and verify it resolves to the address as expected (with loading state).
5. Enter `offchainexample.eth` as a send address and verify you are prompted to enable ENS offchain lookup.
6. Tap `Use ENS Domain` and verify it resolves to the address as expected (with loading state).
7. Navigate to wallet/web3 settings and verify `Allow ENS Offchain Lookup` is shown as `Enabled`
8. Navigate back to send token view
9. Enter `offchainexample.eth` as a send address and verify you are no longer prompted to enable ENS offchain lookup.
10. Navigate to wallet/web3 settings and verify `Allow ENS Offchain Lookup` is shown as `Disabled`
11. Navigate back to send token view
12.  Enter `offchainexample.eth` as a send address and verify it does not resolve the address


## Screenshots:

https://user-images.githubusercontent.com/5314553/221919341-69fdb98d-f176-4732-9afb-eb78ed1df74e.mp4


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
